### PR TITLE
Extract CM_Get_DevNode_PropertyW() and CM_Get_Device_Interface_PropertyW() calls to separate functions

### PR DESCRIPTION
**Pull request by DJm00n**

### CODE REVIEW

**Comment by Youw** [comment]
Please make the change on top of https://github.com/libusb/hidapi/pull/369

File: windows/hid.c [hunk 1]
```diff
@@ -370,46 +383,85 @@ int HID_API_EXPORT hid_exit(void)
...
 	DEVPROPTYPE property_type;
+	wchar_t* property_value = NULL;
 
-	static DEVPROPKEY DEVPKEY_NAME = { { 0xb725f130, 0x47ef, 0x101a, 0xa5, 0xf1, 0x02, 0x60, 0x8c, 0x9e, 0xeb, 0xac }, 10 }; // DEVPROP_TYPE_STRING
-	static DEVPROPKEY PKEY_DeviceInterface_Bluetooth_DeviceAddress = { { 0x2bd67d8b, 0x8beb, 0x48d5, 0x87, 0xe0, 0x6c, 0xda, 0x34, 0x28, 0x04, 0x0a }, 1 }; // DEVPROP_TYPE_STRING
-	static DEVPROPKEY PKEY_DeviceInterface_Bluetooth_Manufacturer = { { 0x2bd67d8b, 0x8beb, 0x48d5, 0x87, 0xe0, 0x6c, 0xda, 0x34, 0x28, 0x04, 0x0a }, 4 }; // DEVPROP_TYPE_STRING
+#ifndef HIDAPI_USE_DDK
+	if (!CM_Get_DevNode_PropertyW)
+		return NULL;
+#endif
```
Comment (Youw) [inline windows/hid.c:396]: after #369 this check is not required

File: windows/hid.c [hunk 2]
```diff
@@ -370,46 +383,85 @@ int HID_API_EXPORT hid_exit(void)
...
+{
+	ULONG len = 0;
+	CONFIGRET cr;
+	DEVPROPTYPE property_type;
+	wchar_t* property_value = NULL;
+
+#ifndef HIDAPI_USE_DDK
+	if (!CM_Get_Device_Interface_PropertyW)
+		return NULL;
+#endif
```
Comment (Youw) [inline windows/hid.c:422]: ditto

File: windows/hid.c [hunk 3]
```diff
@@ -441,31 +493,18 @@ static int hid_internal_get_interface_number(const wchar_t* hardware_id)
...
-
 #ifndef HIDAPI_USE_DDK
-	if (!CM_Get_Device_Interface_PropertyW ||
-		!CM_Locate_DevNodeW ||
-		!CM_Get_Parent ||
-		!CM_Get_DevNode_PropertyW)
+	if (!CM_Locate_DevNodeW ||
+		!CM_Get_Parent)
 		goto end;
 #endif
```
Comment (Youw) [inline windows/hid.c:503]: ditto

**Review by Youw — COMMENTED** [review]
Overall nice update, thanks.

File: windows/hid.c [hunk 4]
```diff
@@ -302,46 +302,74 @@ int HID_API_EXPORT hid_exit(void)
 	return 0;
 }
 
-static void hid_internal_get_ble_info(struct hid_device_info* dev, DEVINST dev_node)
+static wchar_t* hid_internal_get_devnode_property(DEVINST dev_node, const DEVPROPKEY* property_key, DEVPROPTYPE expected_property_type)
 {
```
Comment (Youw) [inline windows/hid.c:306]:
```suggestion
static wchar_t* hid_internal_get_wchar_t_devnode_property(DEVINST dev_node, const DEVPROPKEY* property_key, DEVPROPTYPE expected_property_type)
{
	if (expected_property_type != DEVPROP_TYPE_STRING || expected_property_type != DEVPROP_TYPE_STRING_LIST)
		return NULL;
```

File: windows/hid.c [hunk 5]
```diff
@@ -302,46 +302,74 @@ int HID_API_EXPORT hid_exit(void)
...
+	cr = CM_Get_DevNode_PropertyW(dev_node, property_key, &property_type, (PBYTE)property_value, &len, 0);
+	if (cr != CR_SUCCESS) {
+		free(property_value);
+		return NULL;
+	}
+
+	return property_value;
+}
+
+static wchar_t* hid_internal_get_device_interface_property(const wchar_t* interface_path, const DEVPROPKEY* property_key, DEVPROPTYPE expected_property_type)
```
Comment (Youw) [inline windows/hid.c:326]:
```suggestion
static wchar_t* hid_internal_get_wchar_t_device_interface_property(const wchar_t* interface_path, const DEVPROPKEY* property_key, DEVPROPTYPE expected_property_type)
	if (expected_property_type != DEVPROP_TYPE_STRING || expected_property_type != DEVPROP_TYPE_STRING_LIST)
		return NULL;
```

**Review by Youw — APPROVED** [review]

File: windows/hid.c (hunk 4 above)
Comment (DJm00n) [reply windows/hid.c:306]: there is `property_type != expected_property_type` check in that method already.
what if I just change `wchar_t*` to `PVOID` return type?

Comment (Youw) [reply windows/hid.c:306]: That's one way of doing it.
But you'd have to cast the return value to `wchar_t *`  everywhere you use it.

maybe have both: generic `hid_internal_get_devnode_property` that returns `PVOID` _and_ `hid_internal_get_wchar_t_devnode_property` that does the check suggested above and returns `wchar_t`?

**Pull request merged**